### PR TITLE
:sparkles: add last seen user in top bar

### PR DESCRIPTION
**Pull request by benjaminbrgs**
Closes https://github.com/Zenika/skillZ/issues/319

Actually, the last_seen field in Auth0 is not available.
What I'm doing is the following :
- When the user is connected, I check every 60 seconds the difference in minute between last_seen and current date.
- If difference > 10, update the last_seen value for the current user

### CODE REVIEW

**Comment by PierreBrethes** [comment]
Very good code ! But i wonder why such a big complexity just for a last seen. In my opinion last seen = last login, shouldn't it be only updatable when the user log in ? Et once you have it when he logs, just on the front you compare this date to the current date. 

Maybe i don't have all the context, but still great code with useContext :)

File: src/components/TopBar.tsx [hunk 1]
```diff
@@ -243,10 +245,13 @@ const TopBar = ({ togglePanel }: TopBarProps) => {
...
                   {userAgencyResult?.UserLatestAgency[0]?.agency && (
-                    <span>
+                    <p className={"text-sm"}>
                       {`Zenika ${userAgencyResult?.UserLatestAgency[0]?.agency}`}
-                    </span>
+                    </p>
                   )}
+                  <p className={"text-xs"}>{`${t(
+                    "myProfile.lastSeen"
+                  )} : ${lastSeen.toLocaleString("fr-FR")}`}</p>
```
Comment (PierreBrethes) [inline src/components/TopBar.tsx:254]: are the singapore users using Skillz ? aha

File: src/providers/ActivityProvider.tsx [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+};
+
+export default ActivityProvider;
+
+export const useActivity = (): ActivityProviderType => {
+  const context = useContext<ActivityProviderType | undefined>(
+    ActivityProviderContext
+  );
+  if (!context) {
+    throw new Error("useActivity must be used within a ActivityProvider");
```
Comment (PierreBrethes) [inline src/providers/ActivityProvider.tsx:70]: a => an ActivityPro...

File: src/components/TopBar.tsx (hunk 1 above)
Comment (PierreBrethes) [reply src/components/TopBar.tsx:254]: Good luck to fix this then aha, sorry Benjamin ! But yes, this should adapt to where we use Skillz 😬

Comment (benjaminbrgs) [reply src/components/TopBar.tsx]: It is now fixed :))

**Comment by jeanphibaconnais** [comment]
Hello ! 
In the issue, you spoke about the attribute last_login. And in this PR, you speak about last_seen 🤔  The last_login is not usable ? 

`When the user is connected, I check every 60 seconds the difference in minute between last_seen and current date`. It's not a little hard to do that ? If a script run frequently, could we have less execution ? In this case I think we can increase delay

**Comment by PierreBrethes** [comment]
> Hello ! In the issue, you spoke about the attribute last_login. And in this PR, you speak about last_seen 🤔 The last_login is not usable ?
> 
> `When the user is connected, I check every 60 seconds the difference in minute between last_seen and current date`. It's not a little hard to do that ? If a script run frequently, could we have less execution ? In this case I think we can increase delay

I think it should be better if we

**Comment by MailyLehoux** [comment]
To test : 
```
npm hasura metadata apply && npm generate:local
```
Render : 

Capture d’écran de 2022-11-14 11-46-20

It can be interesting to add your rules in `doc/rg.md` 

And personnally I like this feature :+1: 

With @StephanieHP, we imagined a message with the date of last connection which was displayed only if it had been a while since we had connected. We didn't define a time limit. Maybe we should wait for her return on this POC to reconfirm a larger delay with her?

**Comment by benjaminbrgs** [comment]
ℹ️ My last commit (https://github.com/Zenika/skillZ/pull/328/commits/f6d014051a07ec384c6255c0081bf1169fd9db0e) changes the way to save user's last login in database.

It is now working like the following :
- At each login, two properties are updated in database :
  - current_login = Date.now()
  - last_login = previous value of current_login
- Then, only last_login is displayed in top bar

**Comment by StephanieHP** [comment]
Hello @MailyLehoux and @benjaminbrgs :) 
Can you maybe use the same Typo that is used up for the name of the consultant "Mai-Ly LEHOUX" ? I think it's going to be better.

**Comment by benjaminbrgs** [comment]
> Hello @MailyLehoux and @benjaminbrgs :) Can you maybe use the same Typo that is used up for the name of the consultant "Mai-Ly LEHOUX" ? I think it's going to be better.

Hi @StephanieHP :)
Actually, it is the same typo as the consultant's name but in a different size
Name : 16px
Agency : 14px
Last login : 12px

**Review by PierreBrethes — APPROVED** [review]

File: hasura/metadata/tables.yaml [hunk 3]
```diff
@@ -281,16 +281,20 @@
           email:
             _eq: x-hasura-user-email
         columns:
-          - email
           - botNotifications
+          - current_login
+          - email
+          - last_login
```
Comment (MailyLehoux) [inline hasura/metadata/tables.yaml:287]: `"reason": "Inconsistent object: in table \"User\": in permission for role \"world\": column \"last_login\" does not exist"`

In Hasura the column is called "last_seen" and also, wordl has not select permission on last_seen and current_login
Capture d’écran de 2022-11-29 10-42-56

**Review by MailyLehoux — CHANGES_REQUESTED** [review]

File: hasura/metadata/tables.yaml (hunk 3 above)
Comment (benjaminbrgs) [reply hasura/metadata/tables.yaml:287]: It's weird, did you apply all migrations ? Because on my side, I've the right name and permissions

Comment (MailyLehoux) [reply hasura/metadata/tables.yaml:287]: Yes i've done : 
`npm run hasura migrate apply && npm run hasura metadata apply`

But the message above was displayed during the metadata apply. 

So I tried to stop and rm my docker volumes then apply the seeds to get the column. I don't know why. 

Now it works, but I had to reset all my local datas (wtf ?)

**Review by MailyLehoux — APPROVED** [review]
I approve this PR, we can test it in stagging to be sure that's really works. Maybe it came from my configuration

**Comment by benjaminbrgs** [comment]
@MailyLehoux It seems to work on staging, I let you test as well 👍🏼

**Comment by MailyLehoux** [comment]
nice it works indeed :+1:

**Pull request merged**